### PR TITLE
zephyr: udp throughput fail when legacy roaming

### PR DESCRIPTION
**Pull request by GaofengZhangNXP**
zperf udp tx throughput uses fixed ip address, if run legacy roaming at
this time, roaming will restart dhcp and ip address is changed,
so zperf can't find ip address when zperf send udp packet during legacy
roaming, and zperf udp tx fail.
we think that legacy roaming shouldn't restart dhcp, so we need to
determine whether current connection is a normal connection or a
roaming, and determine whether we should restart dhcp.
and we use assoc_freq parameter in wps_supplicant to get current connect
state.
if assoc_freq is not 0, it indicates that the sta already connects, and
current connection is roaming, so we doesn't start dhcp.
if assoc_freq is 0, it indicates that current connection is a normal
connection rather than roaming, so we want to start dhcp.

### CODE REVIEW

**Comment by krish2718** [comment]
Please raise a manifest PR too.

**Comment by GaofengZhangNXP** [comment]
> Please raise a manifest PR too.
manifest PR: 
https://github.com/zephyrproject-rtos/zephyr/pull/82587

File: src/drivers/driver_zephyr.c [hunk 1]
```diff
@@ -538,6 +538,9 @@ void wpa_drv_zep_event_proc_assoc_resp(struct zep_drv_if_ctx *if_ctx,
 		if (if_ctx->ft_roaming) {
```
Comment (jukkar) [inline src/drivers/driver_zephyr.c:538]: @GaofengZhangNXP please re-write the commit message with proper English, it is very difficult to understand what it is saying. Also please start the sentences with capital letter so that it is easier to separate the sentences.
So the commit message should say the reasoning why this change is being done, this is not clear from the message.

Comment (GaofengZhangNXP) [reply src/drivers/driver_zephyr.c:538]: @jukkar I have updated the commit, please help review it, thanks

**Review by krish2718 — APPROVED** [review]

**Comment by GaofengZhangNXP** [comment]
Hi @jukkar , please help review the patch, if you have not any other suggestion, please help merge it, thank you very much.

**Comment by gannicus-gj** [comment]
Hi @jukkar 
could you pls take time to review before Christmas holiday?

**Review by jukkar — APPROVED** [review]
